swap_meet/vendor.py

```diff
@@ -1,2 +1,62 @@
 class Vendor:
-    pass
\ No newline at end of file
+    '''Creates a Vendor class with an inventory attribute. Inventory is an empty list by default or 
+    you can optionally pass in a list with the keyword inventory'''
+    def __init__(self, inventory=None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        '''Adds an item to a vendor's inventory'''
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        '''Removes an item from a vendor's inventory'''
+        if item not in self.inventory:
+            return False
+        self.inventory.remove(item)
+        return item
+
+    def get_by_category(self, category):
+        "Returns a list of items of a category in a vendor's inventory"
+        item_list = [item for item in self.inventory if item.category == category]
+        return item_list
+
+    def swap_items(self, vendor, my_item, their_item):
+        '''Swaps items (my_item and their_item) between two vendor inventories (self and vendor)'''
+        if my_item not in self.inventory or their_item not in vendor.inventory:
+            return False
+        self.inventory.remove(my_item)
+        self.inventory.append(their_item)
+        vendor.inventory.remove(their_item)
+        vendor.inventory.append(my_item)
+        return True
+
+    def swap_first_item(self, vendor):
+        '''Swaps the first item in two vendor inventories (self and vendor)'''
+        if self.inventory and vendor.inventory:
+            return self.swap_items(vendor, self.inventory[0], vendor.inventory[0])
```

Comment:
Love how you re-use your earlier method! One small bug here: according to the README this method should return `False` if either of the inventories is empty, but it actually returns `None`.